src/idl_gen_dlang.cpp

```diff
@@ -0,0 +1,706 @@
+                   namespace_general += *it;
+                   if (namespace_general.length()) {
+                       namespace_general += ".";
+                   }
+               }
+               std::list<std::string> imports;
+               for (auto it = struct_def.fields.vec.begin();
+                    it != struct_def.fields.vec.end();
+                    ++it) {
+                   auto &field = **it;
+                   if (field.deprecated) continue;
+
+                   if (field.value.type.struct_def)
+                       imports.push_back(field.value.type.struct_def->name);
+                   if (field.value.type.enum_def)
+                       imports.push_back(field.value.type.enum_def->name);
+               }
+               imports.unique();
+               for (auto it = imports.begin(); it != imports.end(); ++it) {
+                   auto import = "import " + namespace_general + (*it) + ";\n";
+                   code += import;
+               }
+       }
+
+       GenComment(struct_def.doc_comment, code_ptr);
+
+       code += "struct " + struct_def.name;
+
+       code += " {\n";
+       //Dlang mixin Struct
+       code += std::string("\tmixin ") + (struct_def.fixed? "Struct" : "Table") + "!" + struct_def.name + ";\n\n";
+
+       if (!struct_def.fixed) {
+           // Generate a special accessor for the table that when used as the root
+           // of a FlatBuffer
+           std::string method_name =  "getRootAs" + struct_def.name;
+           // create convenience method that doesn't require an existing object
+           code += std::string("  ")  + "static " + struct_def.name + " " + method_name + "(ByteBuffer _bb) ";
+           code += "{  return " + struct_def.name + ".init_(_bb.get!int(_bb.position()) + _bb.position(), _bb); }\n";
+
+           if (parser_.root_struct_def_ == &struct_def) {
+               if (parser_.file_identifier_.length()) {
+                   // Check if a buffer has the identifier.
+                   code += std::string("  ")  + "static ";
+                   code += "bool" + struct_def.name;
+                   code += "BufferHasIdentifier(ByteBuffer _bb) { return ";
+                   code += "__has_identifier(_bb, \"" + parser_.file_identifier_;
+                   code += "\"); }\n";
+               }
+           }
+       }
+
+       for (auto it = struct_def.fields.vec.begin();
+            it != struct_def.fields.vec.end();
+            ++it) {
+           auto &field = **it;
+           if (field.deprecated) continue;
+           GenComment(field.doc_comment, code_ptr, "  ");
+           std::string type_name = GenTypeGet(field.value.type);
+           std::string type_name_dest = GenTypeGet(field.value.type);
+           std::string method_start = std::string("") + type_name_dest + " ";
+
+
+           if (field.value.type.base_type == BASE_TYPE_VECTOR) {
+                 method_start = std::string("\t") + "auto " + MakeCamel(field.name, false);
```

Comment:
What is the "false" in the MakeCamel. It is not obvious from the caller side.

Use a two separate methods, an enum, or document argument, like this: `MakeCamel(field.name, /*sss=*/false)`

Where sss is name of the argument actually.